public/docs/ts/latest/guide/style-guide.jade

```diff
@@ -0,0 +1,672 @@
+        message: 'May the Force be with You'
+      };
+      private _toastElement: any;
+
+      // public properties
+      title: string;
+      message: string;
+
+      // ctor
+      constructor(toastService: ToastService) {
+        toastService.activate = this.activate.bind(this);
+      }
+
+      // public functions
+      activate(message = this._defaults.message, title = this._defaults.title) {
+        this.title = title;
+        this.message = message;
+        this._show();
+      }
+
+      ngOnInit() {
+        this._toastElement = document.getElementById('toast');
+      }
+
+      // private functions
+      private _show() {
+        console.log(this.message);
+        this._toastElement.style.opacity = 1;
+        this._toastElement.style.zIndex = 9999;
+
+        window.setTimeout(() => this._hide(), 2500);
+      }
+
+      private _hide() {
+        this._toastElement.style.opacity = 0;
+        window.setTimeout(() => this._toastElement.style.zIndex = 0, 400);
+      }
+    }
+    ```
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ### Defer Logic to Services
+  <a id="035"></a>
+  #### Style 035
+
+    - Defer logic in a component by delegating to services.
+
+    - Define a component for a view, and try not to reuse the component for other views. Instead, move reusable logic to factories and keep the component simple and focused on its view.
+
+      **Why?** Logic may be reused by multiple components when placed within a service and exposed via a function.
+
+      **Why?** Logic in a service can more easily be isolated in a unit test, while the calling logic in the component can be easily mocked.
+
+      **Why?** Removes dependencies and hides implementation details from the component.
+
+      **Why?** Keeps the component slim, trim, and focused.
+
+  ```typescript
+  // avoid
+  export class SessionListComponent implements OnInit {
+    sessions: Session[];
+
+    constructor(private _http: Http) { }
+
+    getSessions() {
+      this.sessions = [];
+      this._http.get(sessionsUrl)
+        .map((response: Response) => <Session[]>response.json().data)
+        .catch(this._exceptionService.catchBadResponse)
+        .finally(() => this._spinnerService.hide())
+        .subscribe(sessions => this.sessions = sessions);
+    }
+    
+
+    ngOnInit() {
+      this.getSessions();
+    }
+  }
+  ```
+    
+  ```typescript
+  // recommended
+  export class SessionListComponent implements OnInit {
+    sessions: Session[];
+
+    constructor(private _sessionService: SessionService) { }
+
+    getSessions() {
+      this.sessions = [];
+      this._sessionService.getSessions()
+        .subscribe(sessions => this.sessions = sessions);
+    }
+
+    ngOnInit() {
+      this.getSessions();
+    }
+  }
+  ```
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ## Services
+
+  ### Singletons
+  <a id="040"></a>
+  #### Style 040
+
+    - Services are singletons and should be used for sharing data and functionality.
```

Comment:
yep. just hadnt figured out how to articulate this, and what is applicable to the guide, if at all. because that depth belongs in the docs, so some of this may just go